src/components/BalanceDistributionBar.js

```diff
@@ -2,68 +2,67 @@
 /* global LoginFileConfig */
 /* global IqonHash */
 /* global BitcoinUtils */
+/* global PolygonUtils */
+/* global EuroUtils */
+
+/** @typedef {{address: string, balance: number, active: boolean, newBalance: number}} Segment */
+/** @typedef {'NIM' | 'BTC' | 'USDC' | 'EUR'} Asset */
 
 class BalanceDistributionBar { // eslint-disable-line no-unused-vars
     /**
      * @param {{
-     *  nimiqAddresses: { address: string, balance: number }[],
-     *  bitcoinAccount: { balance: number },
-     *  swapNimAddress: string,
-     *  nimFiatRate: number,
-     *  btcFiatRate: number,
-     *  newNimBalance: number,
-     *  newBtcBalanceFiat: number,
+     *  leftAsset: Asset,
+     *  rightAsset: Asset,
+     *  leftSegments: Segment[],
+     *  rightSegments: Segment[],
+     *  leftFiatRate: number,
+     *  rightFiatRate: number,
      * }} settings
      * @param {HTMLDivElement} [$el]
      */
     constructor(settings, $el) {
         this.$el = BalanceDistributionBar._createElement($el);
 
         const {
-            nimiqAddresses,
-            bitcoinAccount,
-            swapNimAddress,
-            nimFiatRate,
-            btcFiatRate,
-            newNimBalance,
-            newBtcBalanceFiat,
+            leftAsset,
+            rightAsset,
+            leftSegments,
+            rightSegments,
+            leftFiatRate,
+            rightFiatRate,
         } = settings;
 
-        const nimDistributionData = nimiqAddresses.map(addressInfo => {
-            const active = swapNimAddress === addressInfo.address;
-            const backgroundClass = LoginFileConfig[IqonHash.getBackgroundColorIndex(addressInfo.address)]
-                .className;
-            const oldBalance = Nimiq.Policy.lunasToCoins(addressInfo.balance) * nimFiatRate;
-            const newBalance = active
-                ? Nimiq.Policy.lunasToCoins(newNimBalance) * nimFiatRate
-                : oldBalance;
-
-            return {
-                oldBalance,
-                newBalance,
-                backgroundClass,
-                active,
-            };
-        });
+        const leftDistributionData = leftSegments.map(segment => ({
+            oldBalance: this._unitsToFiat(segment.balance, leftAsset, leftFiatRate),
+            newBalance: this._unitsToFiat(segment.newBalance, leftAsset, leftFiatRate),
+            backgroundClass: leftAsset === 'NIM'
+                ? LoginFileConfig[IqonHash.getBackgroundColorIndex(segment.address)].className
+                : segment.address,
```

Comment:
```suggestion
                : leftAsset.toLowerCase(),
```

While the `address` is set to the currency, this is definitely confusing code. Better to use `leftAsset`.

Similar for `rightDistributionData` below.